src/Faker/Generator.php

```diff
@@ -204,6 +212,38 @@ class Generator
     protected $providers = [];
     protected $formatters = [];
 
+    /**
+     * @var ContainerInterface
+     */
+    private $container;
+
+    public function __construct(ContainerInterface $container = null)
+    {
+        $this->container = $container ?? ContainerBuilder::getDefault();
+    }
+
+    /**
+     * @template T of Extension
+     * @param class-string<T> $id
+     * @return T
+     *
+     * @throws ExtensionNotFound
+     * @throws ContainerExceptionInterface
+     */
+    public function ext(string $id): Extension
+    {
+        if (!$this->container->has($id)) {
+            throw new ExtensionNotFound(sprintf('No Faker extension with id "%s" was loaded.', $id));
+        }
+
+        $extension = $this->container->get($id);
+        if ($extension instanceof GeneratorAwareExtension) {
+            $extension = $extension->withGenerator($this);
```

Comment:
```suggestion
            return $extension->withGenerator($this);
```

Reply:
Same same

Reply:
I reverted this. PHPStan and Psalm got super upset.